ggshield/scan/scanner.py

```diff
@@ -68,15 +68,40 @@ def from_exception(exc: Exception) -> "Results":
         return Results(results=[], errors=[error])
 
 
-class ScanCollection(NamedTuple):
+class ScanCollection:
     id: str
     type: str
     results: Optional[Results] = None
-    scans: Optional[List["ScanCollection"]] = None  # type: ignore[misc]
+    scans: Optional[List["ScanCollection"]] = None
     iac_result: Optional[IaCScanResult] = None
     optional_header: Optional[str] = None  # To be printed in Text Output
     extra_info: Optional[Dict[str, str]] = None  # To be included in JSON Output
 
+    def __init__(
+        self,
+        id: str,
+        type: str,
+        results: Optional[Results] = None,
+        scans: Optional[List["ScanCollection"]] = None,
+        iac_result: Optional[IaCScanResult] = None,
+        optional_header: Optional[str] = None,
+        extra_info: Optional[Dict[str, str]] = None,
+    ):
+        self.id = id
+        self.type = type
+        self.results = results
+        self.scans = scans
+        self.iac_result = iac_result
+        self.optional_header = optional_header
+        self.extra_info = extra_info
+
+        self.has_new_secrets = self._init_has_new_secrets()
+        (
+            self.known_secrets_count,
+            self.new_secrets_count,
```

Comment:
Looks like new_secrets_count does not need to be a member: it's only used to set `has_new_secrets_all_scans`. Either turn it into a member variable, or turn `has_new_secrets_all_scans` into a property returning `self.new_secrets_count > 0`.